aws/ec2_test.go

```diff
@@ -222,3 +243,438 @@ func TestNukeInstances(t *testing.T) {
+		subnets = append(subnets, &ec2.Subnet{SubnetId: awsgo.String(subnetId)})
+	}
+	return &ec2.DescribeSubnetsOutput{Subnets: subnets}
+}
+
+func getDeleteSubnetInput(subnetId string) *ec2.DeleteSubnetInput {
+	return &ec2.DeleteSubnetInput{
+		SubnetId: awsgo.String(subnetId),
+	}
+}
+
+func getDescribeRouteTablesInput(vpcId string) *ec2.DescribeRouteTablesInput {
+	return &ec2.DescribeRouteTablesInput{
+		Filters: []*ec2.Filter{
+			&ec2.Filter{
+				Name:   awsgo.String("vpc-id"),
+				Values: []*string{awsgo.String(vpcId)},
+			},
+		},
+	}
+}
+
+func getDescribeRouteTablesOutput(routeTableIds []string) *ec2.DescribeRouteTablesOutput {
+	var routeTables []*ec2.RouteTable
+	for _, routeTableId := range routeTableIds {
+		routeTables = append(routeTables, &ec2.RouteTable{RouteTableId: awsgo.String(routeTableId)})
+	}
+	return &ec2.DescribeRouteTablesOutput{RouteTables: routeTables}
+}
+
+func getDeleteRouteTableInput(routeTableId string) *ec2.DeleteRouteTableInput {
+	return &ec2.DeleteRouteTableInput{
+		RouteTableId: awsgo.String(routeTableId),
+	}
+}
+
+func getDescribeNetworkAclsInput(vpcId string) *ec2.DescribeNetworkAclsInput {
+	return &ec2.DescribeNetworkAclsInput{
+		Filters: []*ec2.Filter{
+			&ec2.Filter{
+				Name:   awsgo.String("default"),
+				Values: []*string{awsgo.String("false")},
+			},
+			&ec2.Filter{
+				Name:   awsgo.String("vpc-id"),
+				Values: []*string{awsgo.String(vpcId)},
+			},
+		},
+	}
+}
+
+func getDescribeNetworkAclsOutput(networkAclIds []string) *ec2.DescribeNetworkAclsOutput {
+	var networkAcls []*ec2.NetworkAcl
+	for _, networkAclId := range networkAclIds {
+		networkAcls = append(networkAcls, &ec2.NetworkAcl{NetworkAclId: awsgo.String(networkAclId)})
+	}
+	return &ec2.DescribeNetworkAclsOutput{NetworkAcls: networkAcls}
+}
+
+func getDeleteNetworkAclInput(networkAclId string) *ec2.DeleteNetworkAclInput {
+	return &ec2.DeleteNetworkAclInput{
+		NetworkAclId: awsgo.String(networkAclId),
+	}
+}
+
+func getDescribeSecurityGroupsInput(vpcId string) *ec2.DescribeSecurityGroupsInput {
+	return &ec2.DescribeSecurityGroupsInput{
+		Filters: []*ec2.Filter{
+			&ec2.Filter{
+				Name:   awsgo.String("vpc-id"),
+				Values: []*string{awsgo.String(vpcId)},
+			},
+		},
+	}
+}
+
+func getDescribeSecurityGroupsOutput(securityGroupIds []string) *ec2.DescribeSecurityGroupsOutput {
+	var securityGroups []*ec2.SecurityGroup
+	for _, securityGroup := range securityGroupIds {
+		securityGroups = append(securityGroups, &ec2.SecurityGroup{
+			GroupId:   awsgo.String(securityGroup),
+			GroupName: awsgo.String(""),
+		})
+	}
+	return &ec2.DescribeSecurityGroupsOutput{SecurityGroups: securityGroups}
+}
+
+func getDeleteSecurityGroupInput(securityGroupId string) *ec2.DeleteSecurityGroupInput {
+	return &ec2.DeleteSecurityGroupInput{
+		GroupId: awsgo.String(securityGroupId),
+	}
+}
+
+func getDeleteVpcInput(vpcId string) *ec2.DeleteVpcInput {
+	return &ec2.DeleteVpcInput{
+		VpcId: awsgo.String(vpcId),
+	}
+}
```

Comment:
Maybe extract these helpers into a `mock_ec2_utils_for_test.go` file? (the name is a hack to ensure it is only compiled and added to the binary during testing).

Reply:
Sure.